File: pacman-reloaded/src/main/java/config/ConfiguredResources.java
package config;

import com.uqbar.vainilla.appearances.Sprite;
import resource.Resource;

import java.util.HashMap;
import java.util.Map;

public class ConfiguredResources {
	private Map<String, Sprite> sprites;
	private Config config;

	private ConfiguredResources() {
		setSprites(new HashMap<String, Sprite>());
	}

	public ConfiguredResources(Config config) {
		this();
		setConfig(config);
	}

	public Sprite getSprite(String key) {
		return getSprites().get(key);
	}

	public int getInt(String key){
		return getConfig().getInt(key);
	}

	public void loadSprite(String key) {
		getSprites().put(key, Resource.getImage(getConfig().get(key)));
	}

	private Map<String, Sprite> getSprites() {
		return sprites;
	}

	private void setSprites(Map<String, Sprite> sprites) {
		this.sprites = sprites;
	}

	private Config getConfig() {
		return this.config;
	}

	private void setConfig(Config config) {
		this.config = config;
	}
}

File: pacman-reloaded/src/main/java/config/configured_resources.rb
java_import Java::com.uqbar.vainilla.appearances.Sprite
require '../resource/resource'


class ConfiguredResources
  attr_accessor :sprites, :config

  def initialize(config)
    self.sprites = {}
    self.config = config
  end

  def [](key)
    sprites[key]
  end

  def get_int(key)
    self.config.get_int(key)
  end

  def load_sprite(key)
    sprites[key] =  Resource.image self.config[key]
  end
end
